### PR TITLE
105 - ods component - app bar standard

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+package com.orange.ods.demo.ui.components.appbars.top
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+
+@Composable
+fun rememberAppBarTopCustomizationState(
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt:20]: Sometimes we talk about `TopAppBar` and sometimes it's `AppBarTop`. Is there a rule to choose the correct one or should we make all names consistent ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    navigationIconEnabled: MutableState<Boolean> = rememberSaveable { mutableStateOf(true) },
+    actionCount: MutableState<Int> = rememberSaveable { mutableStateOf(1) },
+    overflowMenuEnabled: MutableState<Boolean> = rememberSaveable { mutableStateOf(false) },
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt:23]: Same question as above, should we use values from `TopAppBarConfiguration.defaultConfiguration` as default values ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    val isNavigationIconEnabled: Boolean
+        get() = navigationIconEnabled.value
+
+    val isOverflowMenuEnabled: Boolean
+        get() = overflowMenuEnabled.value
+
+    val isOverflowMenuSwitchEnabled: Boolean
+        get() = actionCount.value <= ComponentAppBarsTop.MaxActionCount - 1
+
+    val maxActionCountSelectable: Int
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt:43]: We could move `ComponentAppBarsTop.MinActionCount` and `ComponentAppBarsTop.MaxActionCount` here and make them private, rename this property to `maxActionCount` and add `minActionCount`. Then we could use `maxActionCount` and `minActionCount` in `ComponentAppBarsTop` (don't know if I'm clear 😅).

File: demo/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -70,6 +70,20 @@
     <string name="component_element_cancel_cross">Cancel cross</string>
 
 
+    <!-- Components App bars: top -->
+    <string name="component_app_bars_top">App bars: top</string>
+    <string name="component_app_bars_top_description">The app bar is a key component and organizes titling, navigation and action buttons.</string>
+    <string name="component_app_bars_top_regular">Regular</string>
+    <string name="component_app_bars_top_title">Page title</string>
```
Comment (florentmaitre) [inline demo/src/main/res/values/strings.xml:77]: This one is unused.

File: docs/components/AppBarsTop.md [hunk 5]
```diff
@@ -69,29 +69,33 @@ OdsTopAppBar(
     title = {
         Text(text = "Title")
     },
+    navigationIcon = {
+        Icon(
+            painter = painterResource(id =  R.drawable.ic_back),
```
Comment (florentmaitre) [inline docs/components/AppBarsTop.md:74]: There is an extra space after the second `=` symbol.

File: lib/src/main/java/com/orange/ods/compose/component/appbar/top/OdsTopAppBar.kt [hunk 6]
```diff
@@ -31,26 +31,37 @@ import androidx.compose.ui.unit.dp
...
 ) {
     TopAppBar(
-        title = title,
+        title = { title?.let { Text(text = title) } },
         modifier = modifier,
-        navigationIcon = navigationIcon,
+        navigationIcon = navigationIcon?.let { navIcon ->
+            {
+                if (onNavigationIconClick != null) {
+                    IconButton(onClick = { onNavigationIconClick() }) {
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/appbar/top/OdsTopAppBar.kt:57]: This line can be slightly simplified to:

```kotlin
IconButton(onClick = onNavigationIconClick) {
```

File: demo/src/main/java/com/orange/ods/demo/ui/OdsDemoTopAppBar.kt [hunk 7]
```diff
@@ -12,78 +12,116 @@ package com.orange.ods.demo.ui
...
+        onNavigationIconClick = upPress,
         actions = {
-            val configuration = LocalConfiguration.current
-            IconButton(onClick = {
-                updateTheme(!configuration.isDarkModeEnabled)
-            }) {
-                if (configuration.isDarkModeEnabled) {
-                    ThemeIcon(iconRes = R.drawable.ic_ui_light_mode, contentDescriptionRes = R.string.theme_changer_icon_content_description_light)
+            val context = LocalContext.current
+            (0 until state.actionCount.value).forEach { index ->
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/OdsDemoTopAppBar.kt:59]: `repeat(state.actionCount.value) { index ->` is a little bit shorter.

File: demo/src/main/java/com/orange/ods/demo/ui/components/Component.kt [hunk 8]
```diff
@@ -40,12 +40,20 @@ sealed class Component(
 
     val imageAlignment: Alignment
         get() = when (this) {
-            BottomNavigation -> Alignment.TopCenter
+            AppBarsTop, BottomNavigation -> Alignment.TopCenter
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/Component.kt:43]: The background is in the provided image, not sure we have to change that. Maybe we can wait for designer review? What do you think ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt (hunk 1 above)
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/appbars/top/AppBarTopCustomizationState.kt:21]: `TopAppBar` is the name of this composable in the sdk so I think we have to follow this naming. But in the DSM we are talking about "App bar: top" and this is how we name it in the ODS Demo app. I propose to keep the package name in the demo app and rename the composables

**Review by florentmaitre — APPROVED** [review]

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**